packages/playwright-test/src/common/testLoader.ts

```diff
@@ -29,64 +27,44 @@ export const defaultTimeout = 30000;
 // we make these maps global.
 const cachedFileSuites = new Map<string, Suite>();
 
-export class TestLoader {
-  private _rootDir: string;
+export async function loadTestFile(file: string, rootDir: string, environment: 'loader' | 'worker', loadErrors: TestError[]): Promise<Suite> {
+  if (cachedFileSuites.has(file))
+    return cachedFileSuites.get(file)!;
+  const suite = new Suite(path.relative(rootDir, file) || path.basename(file), 'file');
+  suite._requireFile = file;
+  suite.location = { file, line: 0, column: 0 };
 
-  constructor(rootDir: string) {
-    this._rootDir = rootDir;
+  setCurrentlyLoadingFileSuite(suite);
+  try {
+    await requireOrImport(file);
+    cachedFileSuites.set(file, suite);
+  } catch (e) {
+    if (environment === 'worker')
```

Comment:
nit: instead of passing `environment` we can make `loadErrors` optional, or provide an error sink.